client/src/components/Schedule/ModalFormEntity.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react';
+import { useRouter } from 'next/router';
+import { Modal, Tabs } from 'antd';
+import FormAddEntity from './FormEntity';
+import useWindowDimensions from '../../utils/useWindowDimensions';
+import { CourseEvent, CourseTaskDetails } from 'services/course';
+import { formatTimezoneToUTC } from 'services/formatter';
+import TaskDetails from './taskDetails';
+import EventDetails from './eventDetails';
+
+const { TabPane } = Tabs;
+
+type Props = {
+  visible: boolean;
+  handleCancel: () => void;
+  courseId: number;
+  typesFromBase: string[];
+  editableRecord?: CourseEvent | null;
+  refreshData: Function;
+};
+
+const ModalFormAddEntity: React.FC<Props> = ({
+  visible,
+  handleCancel,
+  courseId,
+  typesFromBase,
+  editableRecord,
+  refreshData,
+}) => {
+  const router = useRouter();
+  const { course } = router.query;
+  const alias = Array.isArray(course) ? course[0] : course;
+
+  const initialEntityType = (editableRecord && (editableRecord.isTask ? 'task' : 'event')) || 'task';
+  const [entityType, setEntityType] = useState(initialEntityType);
+  const [entityData, setEntityData] = useState(null);
+  const { width } = useWindowDimensions();
+
+  const onFieldsChange = (values: any) => {
+    setEntityData(values);
+  };
+
+  return (
+    <Modal
+      visible={visible}
+      onCancel={handleCancel}
+      width={
```

Comment:
All other modals through the project works perfectly without this prop. Why do we need it here?